src/desktop/components/auth_modal/view.coffee

```diff
@@ -113,6 +119,53 @@ module.exports = class AuthModalView extends ModalView
   toggleMode: (e) ->
     e.preventDefault()
     @state.set 'mode', $(e.target).data('mode')
+  
+  checkAcceptedTerms: () ->
+    input = $('input#accepted_terms_of_service').get(0)
+    input.setCustomValidity? ''
+    if $(input).prop('checked')
+      $('.tos-error').text ''
+      $boxContainer = $('.gdpr-signup__form__checkbox__accept-terms')
+      $boxContainer.attr('data-state', null)
+      true
+    else
+      $boxContainer = $('.gdpr-signup__form__checkbox__accept-terms')
+      $boxContainer.attr('data-state', 'error')
+      input = $('input#accepted_terms_of_service').get(0)
+      input.setCustomValidity('')
+      $('.tos-error').text 'Please agree to our terms to continue'
+      false
+
+  # showTosMessage: () ->
+  #   $boxContainer = $('.gdpr-signup__form__checkbox__accept-terms')
+  #   $boxContainer.attr('data-state', 'error')
+  #   input = $('input#accepted_terms_of_service').get(0)
+  #   input.setCustomValidity('')
+
+
+  fbSignup: (e) ->
+    e.preventDefault()
+    queryData =
+      'signup-intent': @signupIntent
+      'redirect-to': @currentRedirectTo()
+    queryString = $.param(queryData)
+    fbUrl = sd.AP.facebookPath + '?' + queryString
```

Comment:
Ah, kinda cool- rather than baking this into the template you inject it when clicked. Nice!